pipelines/trace_test.go

```diff
@@ -48,16 +47,19 @@ func TestInsecureTrace(t *testing.T) {
 		ReportingPeriod: "24h",
 		Propagators:     []string{"tracecontext", "baggage"},
 	})
-	assert.NoError(t, err)
+	require.NoError(t, err)
 
 	tracer := otel.Tracer("test-library")
-	_, span := tracer.Start(context.Background(), "test-span")
-	span.End()
+	for i := 0; i < 100; i++ {
+		_, span := tracer.Start(context.Background(), ("test-span"))
+		span.End()
+	}
 
 	require.NoError(t, shutdown())
 
 	require.Equal(t, 0, len(server.MetricsRequests()))
 	require.Equal(t, 1, len(server.TraceRequests()))
+	require.Equal(t, 100, spanCount(server))
```

Comment:
This verifies that we send exactly 100 spans when sampling is disabled. I've got the same down in `TestSecureTrace` too; I suppose it could be a separate case if desired.